tests/global_communication/test_communicator.cpp

```diff
@@ -14,22 +14,126 @@ using namespace nest::mc;
 using time_type = float;
 using communicator_type = communication::communicator<time_type, communication::global_policy>;
 
-TEST(communicator, setup) {
-    /*
+bool is_dry_run() {
+    return communication::global_policy::kind() ==
+        communication::global_policy_kind::dryrun;
+}
+
+TEST(communicator, policy_basics) {
     using policy = communication::global_policy;
 
-    auto num_domains = policy::size();
-    auto rank = policy::id();
+    const auto num_domains = policy::size();
+    const auto rank = policy::id();
 
-    auto counts = policy.gather_all(1);
-    EXPECT_EQ(counts.size(), unsigned(num_domains));
-    for(auto i : counts) {
-        EXPECT_EQ(i, 1);
+    EXPECT_EQ(policy::min(rank), 0);
+    if (!is_dry_run()) {
+        EXPECT_EQ(policy::max(rank), num_domains-1);
     }
+}
+
+// some proxy types for communication testing
+struct source_proxy {
+    source_proxy() = default;
+    source_proxy(int g): gid(g) {}
+
+    int gid = 0;
+};
+
+bool operator==(int other, source_proxy s) {return s.gid==other;};
+bool operator==(source_proxy s, int other) {return s.gid==other;};
+
+struct spike_proxy {
+    spike_proxy() = default;
+    spike_proxy(int s, int d): source(s), domain(d) {}
+    source_proxy source = 0;
+    int domain = 0;
+};
+
+TEST(communicator, gather_spikes) {
```

Comment:
I have split it into two tests.
The version that was being used for dry run mode was actually perfectly valid in normal MPI mode too: it handles the case where there were the same number of cells on each rank, and the local cells all generate one spike each.
The other test has a test that returns without running the test if we are in dry run mode.